TRENZ.Extensions.Infisical/InfisicalConfigurationProvider.cs

```diff
@@ -114,6 +129,13 @@ private bool CheckHasChanged()
         return false;
     }
 
+    private void LoadSecretsWithTimeout(Action<IDictionary<string, SecretElement>> callback, TimeSpan timeout)
+    {
+        var task = Task.Run(() => callback(LoadSecrets()));
```

Comment:
this fires a thread and blocks… is that really necessary?

Reply:
Yeah because all APIs force me to provide and load everything sync. From a users perspective, I'd expect my secrets to be available as soon as the call to `builder.AddInfisical()` is done.